lib/clarkson-core-templates.php

```diff
@@ -603,54 +425,71 @@ public static function get_instance() {
 		return $instance;
 	}
 
+	public function add_twig_to_template_hierarchy( array $original_templates ) {
+		$templates = array();
+
+		$directories = array_unique(
+			array(
+				str_replace( get_stylesheet_directory(), '', $this->get_stylesheet_dir() ),
+				str_replace( get_template_directory(), '', $this->get_template_dir() ),
+			)
+		);
+
+		foreach ( $original_templates as $template ) {
+			$pathinfo = pathinfo( $template );
+			foreach ( $directories as $directory ) {
+				$twig_template = $pathinfo['dirname'] . $directory . '/' . $pathinfo['filename'] . '.twig';
```

Comment:
I think this could break when someone does not simply add a path to the current stylesheet/template dir.
If someone does actually filter the clarkson_twig_stylesheet_dir or clarkson_twig_template_dir totally different, this could break.
It's edge-case, but it could be important.

Maybe add something like this:
if nothing was replaced on line 433/434, then just use the result from the filtered path to get the template. Maybe with the realpath() function around it.

Reply:
Can you give me a code example of how this would occur, and steps to reproduce? I'm not sure what you mean.

Reply:
I think that if someone would do this, it may not work correctly:

```
add_filter( 'clarkson_twig_stylesheet_dir', function( $dir ) {
    return WP_CONTENT_DIR . '/templates'
} );
```

But the path i displayed here could be any path not inside the theme folder.

Reply:
Placing template files outside of the theme directory has never worked in Clarkson, because of https://github.com/level-level/Clarkson-Core/blob/3e6b762c620f97073a23d20b0c23337efeb0f8f5/lib/clarkson-core-templates.php#L75.

I'm not sure why you would want to put your template files outside of the theme folder. At that point your probably better of creating your own twig instance, instead of using the one Clarkson Core provides.

Reply:
Fair point. Would also not know why, only thought it was a possibility.